web/src/components/people/AdminView.tsx

```diff
@@ -39,42 +44,93 @@ export default function AdminView() {
     fetchPeople();
   }, []);
 
+  const updatePersonUI = (id: string, training?: TrainingRecord) => {
+    // get people object and find the person with the right id
+    const personToUpdate = people!.find((person) => person.user.id === id);
+
+    if (personToUpdate) {
+      if (training) {
+        if (!personToUpdate.training_record) {
+          personToUpdate.training_record = [];
+        }
+        personToUpdate.training_record.push(training);
+      }
+    }
+  };
+
   if (!people) return null;
 
+  const handleEditTrainingClick = (id: string) => {
+    setId(id);
+    setTrainingDialogOpen(true);
+  };
+
   return (
     <>
+      {trainingDialogOpen && (
+        <TrainingForm id={id} setTrainingDialogOpen={setTrainingDialogOpen} updatePersonUI={updatePersonUI} />
+      )}
+
       <table className={styles.table}>
         <thead>
           <tr>
             <th>User</th>
             <th>Roles</th>
             <th>Agreements</th>
+            <th>Training</th>
           </tr>
         </thead>
-        <tbody>
+        <tbody className={styles.tbody}>
           {people.map((person: Person) => (
             <tr key={person.user.id}>
               <td className={styles.user}>
                 {person.user.username} <small>{person.user.id}</small>
               </td>
-              <td className={styles.roles}>{person.roles}</td>
+              <td className={styles.roles}>{person.roles.join(", ")}</td>
               <td className={styles.agreements}>
                 {person.agreements.confirmed_agreements.map((agreement: ConfirmedAgreement) => (
                   <div key={agreement.agreement_type} className={styles.agreement}>
                     {agreement.agreement_type}
                     {agreement.confirmed_at ? (
-                      <span className={styles.confirmed}>
+                      <span className={styles.valid}>
                         <CheckIcon />
                       </span>
                     ) : (
-                      <span className={styles.unconfirmed}>
+                      <span className={styles.invalid}>
                         <XIcon />
                       </span>
                     )}
                     {agreement.confirmed_at && <small>{convertRFC3339ToDDMMYYYY(agreement.confirmed_at)}</small>}
                   </div>
                 ))}
               </td>
+              <td>
+                <div className={styles.trainingRecord} data-cy="training">
+                  {person.training_record?.map((training: TrainingRecord) => (
+                    <div key={training.kind} className={styles.training}>
+                      {training.kind}
+                      {training.is_valid ? (
+                        <span className={styles.valid}>
+                          <CheckIcon />
+                        </span>
```

Comment:
Not sure if this is a quirk of the test data or not, but I have two nhsd trainings in my user row `portal-jh-admin`. One of them is valid while the other is not, and it won't let me edit this a second time.

Any ideas?

![Screenshot 2025-07-02 at 15 09 04](https://github.com/user-attachments/assets/c207dc07-3327-4ef9-8127-0bdd35fab861)

Reply:
I think the disembodied edit is attached to that training row...I'm just going to have a rejig of this anyway